src/compiler/tsc.ts

```diff
@@ -252,8 +252,11 @@ namespace ts {
         let cachedConfigFileText: string;                           // Cached configuration file text, used for reparsing (if any)
         let configFileWatcher: FileWatcher;                         // Configuration file watcher
         let directoryWatcher: FileWatcher;                          // Directory watcher to monitor source file addition/removal
+        let directoryWatched: string;                               // Where the directory watcher is looking
+        let typeRootsWatched = createMap<boolean>();                // List of watchers of type roots outside the compilation root

```

Comment:
nit: I'd make it a `createMap<true>()` :) since it is effectively a set - otherwise there might be questions if we can ever have `false` value here and what does it mean